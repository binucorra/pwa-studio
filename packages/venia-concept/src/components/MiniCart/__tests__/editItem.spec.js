import React, { useEffect } from 'react';
import { createTestInstance, useQuery } from '@magento/peregrine';

import EditItem from '../editItem';

jest.mock('react', () => {
    const React = jest.requireActual('react');
    return Object.assign(React, { useEffect: jest.fn(React.useEffect) });
});
jest.mock('@magento/peregrine');
jest.mock('../cartOptions');
jest.mock('src/components/LoadingIndicator', () => {
    return {
        __esModule: true,
        loadingIndicator: '( Loading Indicator Component Here )'
    };
});

test('renders null when item not supplied', () => {
    const tree = createTestInstance(
        <EditItem />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders cart options when item has no options', () => {
    const props = {
        item: {
            options: []
        }
    };

    const tree = createTestInstance(
        <EditItem {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders a loading indicator while running query', () => {
    const props = {
        item: {
            options: ['a', 'b', 'c']
        }
    };

    const tree = createTestInstance(
        <EditItem {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders cart options when item has options', () => {
    const queryApi = {
        runQuery: jest.fn(),
        setLoading: jest.fn()
    };
    const queryResult = {
        data: {
            products: {
                items: []
            }
        }
    };
    useQuery.mockReturnValueOnce([ queryResult, queryApi ]);
    useEffect.mockReturnValueOnce(undefined);

    const props = {
        item: {
            options: ['a', 'b', 'c']
        }
    };

    const tree = createTestInstance(
        <EditItem {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
