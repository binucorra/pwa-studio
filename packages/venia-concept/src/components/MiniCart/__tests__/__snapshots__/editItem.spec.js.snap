// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a loading indicator while running query 1`] = `
<span>
  ( Loading Indicator Component Here )
</span>
`;

exports[`renders cart options when item has no options 1`] = `"( Cart Options Component Here )"`;

exports[`renders cart options when item has options 1`] = `"( Cart Options Component Here )"`;

exports[`renders null when item not supplied 1`] = `null`;
